ipc/nano/src/lib.rs

```diff
@@ -54,9 +54,9 @@ impl<S> GuardedSocket<S> where S: WithSocket<Socket> {
 }
 
 impl<S> Deref for GuardedSocket<S> where S: WithSocket<Socket> {
-    type Target = S;
+    type Target = Arc<S>;
 
-    fn deref(&self) -> &S {
+    fn deref(&self) -> &Arc<S> {
```

Comment:
does the implementation do further cloning of `self` and sharing it between threads? if it doesn't then the impl should just be for `T`. that is strictly more flexible.

even so, that doesn't require this `deref` impl